Mathlib/Probability/Integration.lean

```diff
@@ -127,6 +127,27 @@ theorem lintegral_mul_eq_lintegral_mul_lintegral_of_indepFun'' (h_meas_f : AEMea
     ∫⁻ ω, f ω * g ω ∂μ = (∫⁻ ω, f ω ∂μ) * ∫⁻ ω, g ω ∂μ :=
   lintegral_mul_eq_lintegral_mul_lintegral_of_indepFun' h_meas_f h_meas_g h_indep_fun
 
+theorem lintegral_fin_prod_eq_lintegral_fin_prod_lintegral_of_indepFun {ι : Type*} [DecidableEq ι]
+    [IsProbabilityMeasure μ] (s : Finset ι) (X : ι → Ω → ℝ≥0∞)
+    (hX : iIndepFun (fun _ ↦ ENNReal.measurableSpace) X μ)
+    (x_mea : ∀ i, Measurable (X i)) :
+    ∫⁻ ω, ∏ i ∈ s, (X i ω) ∂μ = ∏ i ∈ s, ∫⁻ ω, X i ω ∂μ := by
+  induction s using Finset.induction
```

Comment:
You should be able to remove the `IsProbabilityMeasure` hypothesis, because it is implied by the independence hypothesis. Something like this (untested code):
```suggestion
    (s : Finset ι) (X : ι → Ω → ℝ≥0∞)
    (hX : iIndepFun (fun _ ↦ ENNReal.measurableSpace) X μ)
    (x_mea : ∀ i, Measurable (X i)) :
    ∫⁻ ω, ∏ i ∈ s, (X i ω) ∂μ = ∏ i ∈ s, ∫⁻ ω, X i ω ∂μ := by
  have : IsProbabilityMeasure μ := hX.isProbabilityMeasure
  induction s using Finset.induction
```